src/feature_collection.rs

```diff
@@ -144,3 +160,108 @@ impl<'de> Deserialize<'de> for FeatureCollection {
+
+        let features = iter
+            .into_iter()
+            .inspect(|feat| {
+                // Try to compute the bounding-box
+
+                let (curr_bbox, curr_len) = match &mut bbox {
+                    Some(curr_bbox) => {
+                        let curr_len = curr_bbox.len();
+                        (curr_bbox, curr_len)
+                    }
+                    None => {
+                        // implies we can't compute a
+                        // bounding-box for this collection
+                        return;
+                    }
+                };
+
+                match &feat.bbox {
+                    None => {
+                        bbox = None;
+                    }
+                    Some(fbox) if fbox.len() == 0 || fbox.len() % 2 != 0 => {
+                        bbox = None;
+                    }
+                    Some(fbox) if curr_len == 0 => {
+                        // First iteration: just copy values from fbox
+                        *curr_bbox = fbox.clone();
+                    }
+                    Some(fbox) if curr_len != fbox.len() => {
+                        bbox = None;
+                    }
+                    Some(fbox) => {
+                        // Update bbox by computing min/max
+                        curr_bbox.iter_mut().zip(fbox.iter()).enumerate().for_each(
+                            |(idx, (bc, fc))| {
+                                if idx < curr_len / 2 {
+                                    // These are the min coords
+                                    *bc = fc.min(*bc);
+                                } else {
+                                    *bc = fc.max(*bc);
+                                }
+                            },
+                        );
+                    }
+                };
+            })
+            .collect();
+        FeatureCollection {
+            bbox,
+            features,
+            foreign_members: None,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{Feature, FeatureCollection, Value};
+
+    #[test]
+    fn test_fc_from_iterator() {
+        let features: Vec<Feature> = vec![
+            {
+                let mut feat: Feature = Value::Point(vec![0., 0., 0.]).into();
+                feat.bbox = Some(vec![-1., -1., -1., 1., 1., 1.]);
+                feat
+            },
+            {
+                let mut feat: Feature = Value::MultiPoint(vec![
+                    vec![10., 10., 10.],
+                    vec![11., 11., 11.],
+                ]).into();
+                feat.bbox = Some(vec![10., 10., 10., 11., 11., 11.]);
+                feat
+            },
+        ];
+
+        let fc: FeatureCollection = features.into_iter().collect();
+        assert_eq!(fc.features.len(), 2);
+
+        assert!(fc.bbox.is_some());
+        let bbox = fc.bbox.as_ref().unwrap();
+        assert_eq!(bbox.len(), 6);
+        for (i, coord) in bbox.iter().enumerate() {
```

Comment:
To make this test more declarative, could we equally write something like:

`assert_eq!(bbox, vec![-1., -1., -1, 11., 11., 11.])`

Reply:
Well, I wanted to avoid any exact comparisons with float values.

Reply:
Your suggestion also works, and it does look neater.  I guess with bbox, it's just min/max, so we should be fine even if we change the coords in the future.